text/0000-trait-alias.md

```diff
@@ -150,6 +150,22 @@ more predicates, just like any other `where` clause.
 You cannot directly `impl` a trait alias, but you can have them as *bounds*, *trait objects* and
 *impl Trait*.
 
+----
+
+It is an error to attempt to override a previously specified
+equivalence constraint with a non-equivalent type. For example:
+
+```rust
+trait SharableIterator = Iterator + Sync;
+trait IntIterator = Iterator<Item=i32>;
+
+fn quux1<T: SharableIterator<Item=f64>>(...) { ... } // ok
+fn quux2<T: IntIterator<Item=i32>>(...) { ... } // ok (perhaps subject to lint warning)
```

Comment:
I am happy either way.

Originally I was going to list it as disallowed, as you suggest.

But then I worried that might not be consistent with other design choices. E.g. one can add where clauses that are already implied by pre-existing constraints, right? So should it be an error to add a redundant equivalence constraint?

Maybe best (or at least easiest in terms of getting the RFC merged) to take out the `fn quux2` example, and then make a new Unresolved Question for this case?